subprojects/docs/src/docs/userguide/authoring-builds/introduction/build_lifecycle.adoc

```diff
@@ -0,0 +1,152 @@
+
+Gradle is an example of *dependency-based programming*: you define tasks and dependencies between tasks.
+
+image::author-gradle-1.png[]
+
+Gradle guarantees that these tasks will execute in order of their dependencies.
+Your build scripts and plugins configure this dependency graph.
+
+There are phases Gradle passes through as it interprets those scripts.
+
+== Task Graphs
+
+Gradle builds the task graph *before* executing any task.
+
+Within each project, tasks form a http://en.wikipedia.org/wiki/Directed_acyclic_graph[Directed Acyclic Graph] (DAG).
+
+This diagram shows two example task graphs, one abstract and the other concrete, with dependencies between tasks represented as arrows:
+
+image::task-dag-examples.png[]
+
+[[build_lifecycle_events]]
+Both plugins and build scripts contribute to the task graph via the <<writing_tasks#sec:task_dependencies,task dependency mechanism>>.
+
+[[sec:build_phases]]
+== Build Phases
+
+A Gradle build has three distinct phases.
+
+Gradle runs these phases in order:
+
+PHASE 1. Initialization::
+- Detects the `settings.gradle.kts` file.
+- Creates a link:{groovyDslPath}/org.gradle.api.initialization.Settings.html[`Settings`] instance.
+- Evaluates the settings file to determine which projects (and included builds) make up the build.
+- Creates a link:{groovyDslPath}/org.gradle.api.Project.html[`Project`] instance for every project.
+PHASE 2. Configuration::
+- Evaluates the build scripts (`build.gradle.kts`) of every project participating in the build.
+- Creates a task graph for requested tasks.
+PHASE 3. Execution::
+- Schedules and executes the selected tasks.
+- Dependencies between tasks determine execution order.
+- Execution of tasks can occur in parallel.
+
+image::build-lifecycle-example.png[]
+
+=== Example
+
+The following example shows which parts of settings and build files correspond to various build phases:
+
+====
+include::sample[dir="snippets/buildlifecycle/basic/kotlin",files="settings.gradle.kts[];build.gradle.kts[]"]
+include::sample[dir="snippets/buildlifecycle/basic/groovy",files="settings.gradle[];build.gradle[]"]
+====
+
+The following command configures and executes the `test` and `testBoth` tasks specified above.
+Because Gradle only configures requested tasks and their dependencies, the `configured` task never configures:
+
+[source.multi-language-sample,kotlin]
+----
+> gradle test testBoth
+include::{snippetsPath}/buildlifecycle/basic/tests/buildlifecycle.out[]
+----
+[source.multi-language-sample,groovy]
+----
+> gradle test testBoth
+include::{snippetsPath}/buildlifecycle/basic/tests/buildlifecycle.out[]
+----
+
+[[sec:initialization]]
+== Phase 1. Initialization
+
+In the *initialization phase*, Gradle detects the set of projects (root and subprojects) and included builds participating in the build.
+
+Gradle first evaluates the settings file (`settings.gradle.kts`) and instantiates a `Settings` object.
+Then, Gradle instantiates `Project` instances for each project.
```

Comment:
Does this happen in the initialization phase? I thought it happens in the configuration phase?

Reply:
Project is created in the init phase and "populated" in the config phase.